### PR TITLE
Rainbow Home Screen revamp – 2nd iteration

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-4441
Figma: https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=681%3A20794

## What changed (plus any additional context for devs)

This PR is a continuation of #4106 and implements the second iteration of [designs](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=681%3A20794) based off internal dog-fooding of #4106.

It includes:

- Removing the "Swipe down" for QR scanner on the Discover screen (and making the Discover screen more of a static screen). [Design spec](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=390%3A16577)
- Adding a context menu to the "More" button. [Design spec](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=681%3A21103)
- Adding a context menu to the profile avatar. [Design spec](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=682%3A24773)
- Removed "Copy" as the first action button, and replaced it with "Buy". [Design spec](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=681%3A21103)
- Revamped the QR scanner screen. [Design spec](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=681%3A21897)
- The "Connected Apps" entrypoint is now in the "More" context menu instead of the QR scanner screen. [Design spec](https://www.figma.com/file/XdDSQnZL4cGc1Y8Abvv3EI/Home-Page-Progression?node-id=681%3A21103)

### iOS

https://user-images.githubusercontent.com/7336481/193488247-b6230645-d16c-47b7-a315-fd7d38ae7296.mp4

### Android

https://user-images.githubusercontent.com/7336481/193489520-7344c787-8c2c-4f4f-9b00-897442950e06.mp4

## What to test

- Ensure the functionalities listed in "What's changed" still works across iOS and Android.

### CODE REVIEW

File: src/components/qrcode-scanner/CameraDimmer.js [hunk 1]
```diff
@@ -11,20 +10,12 @@ const Dim = styled(Animated.View)({
 });
 
 export default function CameraDimmer({ children, cameraVisible }) {
-  const scrollPosition = usePagerPosition();
```
Comment (osdnk) [inline src/components/qrcode-scanner/CameraDimmer.js:14]: why?

**Review by osdnk — COMMENTED** [review]
you sure that the header in your PR should appear without any animation or anything? That looks quite rough

cc @mikearndt

**Review by osdnk — COMMENTED** [review]
I approve from the perspective of navigation and RLV. I didn't read yet the whole code, nor run it on the device.

File: src/components/asset-list/RecyclerAssetList2/index.tsx [hunk 2]
```diff
@@ -54,3 +63,35 @@ function RecyclerAssetList({
 }
 
 export default React.memo(RecyclerAssetList);
+
+// //////////////////////////////////////////////////////////
```
Comment (sammdec) [inline src/components/asset-list/RecyclerAssetList2/index.tsx:67]: Shall we delete this?

**Review by sammdec — APPROVED** [review]
Reviewed code and tested on ios and android physical devices

**Review by BrodyHughes — CHANGES_REQUESTED** [review]
Still checking over core functionality, but have noticed an overlap with status bars + our top menu items on Android devices. Screenshots are a Pixel 6 w/ Android 13 and a Samsung S10 with Android 12.
Screenshot_20221007-130629_Rainbow
Screenshot_20221007-130242

**Comment by BrodyHughes** [comment]
The animation between switching wallets as seen [here
](https://github.com/rainbow-me/rainbow/pull/4106) seems to not be working. Also get the add cash screen sometimes before the wallet loads.
https://user-images.githubusercontent.com/41711440/195148547-65b0fe5a-4268-40be-a68d-b946c200b2dc.mov

**Comment by jxom** [comment]
@BrodyHughes – animation seems to work on my device.
the add cash screen is currently in the app store version too, not sure if it is in the scope of this PR.

**Comment by BrodyHughes** [comment]
A few more visual bugs:
1. The shadow on the pfp circle is different on the homescreen vs the txn screen (homescreen is colored, txn screen is black). Transparent backgrounds on pfps also look like they are treated slightly different.
Screen Shot 2022-10-12 at 11 47 15 AM
Screen Shot 2022-10-12 at 11 47 24 AM

2. On Android there is no pull down animation to refresh like there is on iOS (just a loading circle).
https://user-images.githubusercontent.com/41711440/195403013-4ad5d969-84ce-43d1-8df5-c724d225949e.mp4

3. On Android is there supposed to be a wallet switch animation? 
4. On Android the height of the menu items looks better but is still a little high. It overlaps with the camera on some devices.
Screen Shot 2022-10-12 at 11 47 24 AM

**Review by estrattonbailey — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]

**Pull request merged**